janusgraph-core/src/main/java/org/janusgraph/core/ConfiguredGraphFactory.java

```diff
@@ -0,0 +1,163 @@
+            });
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+        configGraphManagement.createConfiguration(new MapConfiguration(templateConfigMap));
+        return g;
+    }
+
+    /**
+     * Open a {@link JanusGraph} using a previously created Configuration using the
+     * {@link ConfigurationManagementGraph} API; if a corresponding configuration does not exist, we throw
+     * a {@link RuntimeException}, else return the Graph;
+     *
+     * NOTE: If your configuration corresponding to this graph does not contain information about
+     * the backend's keyspace/table/storage directory, then we set the keyspace/table to the
+     * graphName or set the storage directory to the storage_root + /graphName.
+     */
+    public static JanusGraph open(String graphName) {
+        final ConfigurationManagementGraph configGraphManagement = getConfigGraphManagementInstance();
+        final Map<String, Object> graphConfigMap = configGraphManagement.getConfiguration(graphName);
+        Preconditions.checkState(null != graphConfigMap,
+                                "Please create configuration for this graph using the ConfigurationManagementGraph#createConfiguration API.");
+        return (JanusGraph) JanusGraphManager.getInstance().openGraph(graphName, (String gName) -> {
+            return new StandardJanusGraph(new GraphDatabaseConfiguration(new CommonsConfiguration(new MapConfiguration(graphConfigMap))));
+        });
+    }
+
+    /**
+     * Get a Set of the graphNames that exist in your configuration management graph
+     *
+     */
+    public static Set<String> getGraphNames() {
+        final ConfigurationManagementGraph configGraphManagement = getConfigGraphManagementInstance();
+        final List<Map<String, Object>> configurations = configGraphManagement.getConfigurations();
+        return configurations.stream().map(elem -> {
+            return (String) elem.getOrDefault(ConfigurationManagementGraph.PROPERTY_GRAPH_NAME, null);
+        }).filter(Objects::nonNull).collect(Collectors.toSet());
+    }
+
+    /**
+     * Removes the graph corresponding to the supplied {@link String} graphName
+     * from the {@link JanusGraphManager} {@link Map<String, Graph} graph reference tracker and
+     * returns the corresponding Graph, or null if it doesn't exist.
+     *
+     * @param configuration Graph
+     * @return JanusGraph
+     */
+    public static JanusGraph close(String graphName) {
+        return (JanusGraph) JanusGraphManager.getInstance().removeGraph(graphName);
+    }
+
+    /**
+     * Returns a {@link MapConfiguration} corresponding to supplied .properties file
+     * which lives at the supplied {@link String} fileLocation.
+     *
+     * @param {@link String} properties file location
+     *
+     * @return MapConfiguration
+     */
+    public static MapConfiguration fileToMapConfiguration(final String fileLocation) throws IOException {
```

Comment:
I think it would be a cleaner API to just provide a generic `Configuration` back here and let clients convert if needed. Only usage I saw was a test case that's easily refactored to support generic Configuration. Also use PropertiesConfiguration to cleanup code

```java
final Configuration config = new PropertiesConfiguration(fileLocation);
```